### ISSUE TITLE
tool for handling large sets of modules

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-framework#561]
We need a tool that makes handling large sets of modules more easy.

A likely candidate is `lmod` (see also #110), although it will need to be enhanced.

Desired features include:
- hiding (sub)toolchains (depends on #560)
- setting a default toolchain to use
- filter `module av` (e.g. `module spider`)
- support for automagically swapping loaded modules, depending on toolchain of module that is being loaded, with different strictness options, e.g.:
  - `loose`: unload already loaded modules that conflict with last specified toolchain
  - `tryswap`: try to swap already loaded modules to adher to last specified toolchain (and fail it the required modules are not there)
  - `strict`: fail if last specified toolchain conflicts with toolchain used by currently loaded moduels

TL;DR: a need a modules tool that is toolchain-aware to make handling large module sets easier
